js/views/modals/wallet/SpendConfirmBox.js

```diff
@@ -21,6 +21,7 @@ export default class extends baseVw {
     };
 
     this.lastFetchFeeEstimateArgs = {};
+    this.metricsOrigin = options.metricsOrigin;
```

Comment:
Can we please handle the possibility of this not being provided? In the current implementation it will probably end up sending something like `undefined_ConfirmBoxSend`.

I'm not sure we should make it required because could there reasonably be a case we'd want a spend confirm box that we don't track via analytics?

Perhaps a less rigid approach would be to make an internal recordEvent helper:

```
recordEvent(..args) {
  if (!this.metricsOrigin) return;
  recordEvent(...args);
}
```